cmd/shell.go

```diff
@@ -28,6 +28,7 @@ import (
 	"fmt"
 	"io/ioutil"
 	"log"
+	"math"
```

Comment:
Honestly, rather than pulling in this whole dependency just for `math.Min()` -- which doesn't even solve your problem without a lot of casting! -- I would prefer that you just define your own "min" function or just use an `if` statement in context.